[Bug Report] When the table component uses tree lazy loading mode, how to report errors if some data has sub-nodes

<!-- generated by https://eleme-issue.surge.sh DO NOT REMOVE -->

### Element UI version
2.7.0

### OS/Browsers version
last 2 chrome version

### Vue version
2.6.10

### Reproduction Link
[https://codepen.io/panhezeng/pen/YMPXpz](https://codepen.io/panhezeng/pen/YMPXpz)

### Steps to reproduce
开启table组件的懒加载模式，treeData数据中，部分节点的子节点数据量大，需要懒加载，设置了hasChildren，部分节点不需要懒加载，直接有children数组。
这时候点击展开箭头，就会报错。[Vue warn]: Error in render: "Error: row is required when get row identity"

具体看https://codepen.io/panhezeng/pen/YMPXpz 重现代码

### What is Expected?
点击展开箭头，正常展开。可以指定部分节点懒加载，部分节点直接展示子节点数据，混用。

### What is actually happening?
点击展开箭头，console报错
[Vue warn]: Error in render: "Error: row is required when get row identity"

found in

---> <ElTableBody>
       <ElTable> at packages/table/src/table.vue
         <Root>
warn @ vue.js:634
logError @ vue.js:1893
globalHandleError @ vue.js:1888
handleError @ vue.js:1848
Vue._render @ vue.js:3547
updateComponent @ vue.js:4061
get @ vue.js:4472
run @ vue.js:4547
flushSchedulerQueue @ vue.js:4305
(anonymous) @ vue.js:1989
flushCallbacks @ vue.js:1915
Promise.then (async)
timerFunc @ vue.js:1942
nextTick @ vue.js:1999
queueWatcher @ vue.js:4397
update @ vue.js:4537
notify @ vue.js:745
mutator @ vue.js:897
(anonymous) @ index.js:1
(anonymous) @ index.js:1
load @ pen.js:79
sn.loadData @ index.js:1
handleTreeExpandIconClick @ index.js:1
invokeWithErrorHandling @ vue.js:1863
invoker @ vue.js:2188
original._wrapper @ vue.js:7541
vue.js:1897 Error: row is required when get row identity
    at Qi (index.js:1)
    at VueComponent.getKeyOfRow (index.js:1)
    at index.js:1
    at Proxy.renderList (vue.js:2637)
    at Proxy.render (index.js:1)
    at VueComponent.Vue._render (vue.js:3545)
    at VueComponent.updateComponent (vue.js:4061)
    at Watcher.get (vue.js:4472)
    at Watcher.run (vue.js:4547)
    at flushSchedulerQueue (vue.js:4305)
logError @ vue.js:1897
globalHandleError @ vue.js:1888
handleError @ vue.js:1848
Vue._render @ vue.js:3547
updateComponent @ vue.js:4061
get @ vue.js:4472
run @ vue.js:4547
flushSchedulerQueue @ vue.js:4305
(anonymous) @ vue.js:1989
flushCallbacks @ vue.js:1915
Promise.then (async)
timerFunc @ vue.js:1942
nextTick @ vue.js:1999
queueWatcher @ vue.js:4397
update @ vue.js:4537
notify @ vue.js:745
mutator @ vue.js:897
(anonymous) @ index.js:1
(anonymous) @ index.js:1
load @ pen.js:79
sn.loadData @ index.js:1
handleTreeExpandIconClick @ index.js:1
invokeWithErrorHandling @ vue.js:1863
invoker @ vue.js:2188
original._wrapper @ vue.js:7541
vue.js:634 [Vue warn]: Error in render: "Error: row is required when get row identity"

<!-- generated by https://eleme-issue.surge.sh DO NOT REMOVE -->

--- a/packages/table/src/table-body.js
+++ b/packages/table/src/table-body.js
@@ -35,7 +35,7 @@ export default {
         prev.push(item);
         const rowKey = this.store.table.getRowKey(item);
         const parent = this.store.states.treeData[rowKey];
-        if (parent && parent.children) {
+        if (parent && parent.children && parent.hasChildren) {
           const tmp = [];
           const traverse = (children) => {
             if (!children) return;

--- a/test/unit/specs/table.spec.js
+++ b/test/unit/specs/table.spec.js
@@ -1,4 +1,4 @@
-import { createVue, triggerEvent, destroyVM } from '../util';
+import { createVue, triggerEvent, destroyVM, waitImmediate } from '../util';
 
 const DELAY = 10;
 const testDataArr = [];
@@ -1956,7 +1956,7 @@ describe('Table', () => {
     }, DELAY);
   });
 
-  it('load substree row data', (done) => {
+  it('load substree row data', async() => {
     const vm = createVue({
       template: `
         <el-table :data="testData" row-key="release" lazy :load="load">
@@ -1968,6 +1968,11 @@ describe('Table', () => {
       `,
       data() {
         const testData = getTestData();
+        testData[testData.length - 1].children = [
+          {
+            name: 'A Bug\'s Life copy 1', release: '2008-1-25-1', director: 'John Lasseter', runtime: 95
+          }
+        ];
         testData[1].hasChildren = true;
         return {
           testData: testData
@@ -1986,14 +1991,15 @@ describe('Table', () => {
         }
       }
     }, true);
-    setTimeout(() => {
-      const expandIcon = vm.$el.querySelector('.el-table__expand-icon');
-      expandIcon.click();
-      setTimeout(() => {
-        expect(expandIcon.classList.contains('el-table__expand-icon--expanded')).to.be.true;
-        expect(vm.$el.querySelectorAll('.el-table__row').length).to.equal(7);
-        done();
-      }, DELAY);
-    }, DELAY);
+
+    await waitImmediate();
+
+    const expandIcon = vm.$el.querySelector('.el-table__expand-icon');
+    expandIcon.click();
+
+    await waitImmediate();
+
+    expect(expandIcon.classList.contains('el-table__expand-icon--expanded')).to.be.true;
+    expect(vm.$el.querySelectorAll('.el-table__row').length).to.equal(8);
   });
 });

